databuilder/databuilder/transformer/complex_type_transformer.py

```diff
@@ -0,0 +1,42 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import importlib
+from typing import Any
+
+from pyhocon import ConfigTree
+
+from databuilder.models.table_metadata import TableMetadata
+from databuilder.transformer.base_transformer import Transformer
+
+PARSING_FUNCTION = 'parsing_function'
+
+
+class ComplexTypeTransformer(Transformer):
+    """
+    Transforms complex types for columns in a table.
+    Takes a TableMetadata object and iterates over the columns.
+    The configured parser takes each column's type and key and sets the
+    column's type_metadata field with the parsed results contained in a
+    TypeMetadata object.
+    """
+    def init(self, conf: ConfigTree) -> None:
+        parsing_function = conf.get_string(PARSING_FUNCTION)
```

Comment:
a check here to make sure `parsing_function` is the right type of function would be good